functions/__sf_section_docker.fish

```diff
@@ -0,0 +1,51 @@
+#
+# Docker
+#
+# Current Docker version and Machine name.
+
+function __sf_section_docker -d "Display docker version and machine name"
+    # ------------------------------------------------------------------------------
+    # Configuration
+    # ------------------------------------------------------------------------------
+
+    __sf_util_set_default SPACEFISH_DOCKER_SHOW true
+    __sf_util_set_default SPACEFISH_DOCKER_PREFIX "is "
+    __sf_util_set_default SPACEFISH_DOCKER_SUFFIX $SPACEFISH_PROMPT_DEFAULT_SUFFIX
+    __sf_util_set_default SPACEFISH_DOCKER_SYMBOL "🐳 "
+    __sf_util_set_default SPACEFISH_DOCKER_COLOR cyan
+    __sf_util_set_default SPACEFISH_DOCKER_VERBOSE_VERSION false
+
+    # ------------------------------------------------------------------------------
+    # Section
+    # ------------------------------------------------------------------------------
+
+    [ $SPACEFISH_DOCKER_SHOW = false ]; and return
+
+    # Show Docker version only if docker is installed
+    type -q docker; or return
+
+    # Show docker version only when pwd has dockerfile or docker-compose.yml or COMPOSE_FILE
+    [ -f ./Dockerfile -o -f ./docker-compose.yml ]; or test "$COMPOSE_FILE"; or return
+
+    # Show Docker version only if docker deamon is up
+    docker version > /dev/null 2>&1
```

Comment:
Sorry if I didn't make myself clear. 😅 
This line check shouldn't be needed altogether if  we go about it in [the same way as spaceship has](https://github.com/denysdovhan/spaceship-prompt/blob/c5ed80a65b8693f98f6fc8f783ce228efd8eb618/sections/docker.zsh#L32-L33):
If `docker_version` doesn't prepend a "v" to the `docker version` command, it should be an empty string if the docker daemon isn't running, causing an early return on line 36.

The "v" could then be added on line 49 between the two variables.